src/dag/read.rs

```diff
@@ -54,3 +54,39 @@ pub async fn get_head(kvr: &dyn kv::Read, name: &str) -> Result<Option<String>>
     }
     Ok(None)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::kv::memstore::MemStore;
+    use crate::kv::Store;
+
+    #[async_std::test]
+    async fn test_has_chunk() {
+        async fn test(hash: &str, del_haskey: bool, expect_has: bool) {
+            let kv = MemStore::new();
+            let kvw = kv.write().await.unwrap();
+            let kvr = kv.read().await.unwrap();
+            let r = Read { kvr };
+
+            kvw.put(&Key::ChunkData("haskey").to_string(), &vec![0u8, 1])
+                .await
+                .unwrap();
+            kvw.commit().await.unwrap();
+            assert_eq!(expect_has, r.has_chunk(&hash).await.unwrap());
+
+            // Mayve test isolation: other txs should not affect what the read tx, which is
```

Comment:
Typo in "Mayve" here.

Reply:
done in https://github.com/rocicorp/repc/pull/41